game-core/src/main/java/games/strategy/triplea/attachments/TerritoryEffectAttachment.java

```diff
@@ -98,6 +101,36 @@ public int getCombatEffect(final UnitType type, final boolean defending) {
     return defending ? combatDefenseEffect.getInt(type) : combatOffenseEffect.getInt(type);
   }
 
+  private void setMovementCostModifier(final String value) throws GameParseException {
```

Comment:
It would be good to see this tested, at least as an integration test that the parsing is good. Perhaps add an example attachment to an XML or see about decomposing the logic so that this can be an isolated unit. Another alternative is to mark this as `@VisibleForTesting` and have the method return value so that it can be tested directly.

Reply:
This is tested indirectly by some of the unit tests that were added. I kind of go back and forth on unit testing all these XML parsing methods as there are a lot of them and most are fairly simple. For now, I'm gonna consider it out of scope given the rest of the methods in this class aren't directly unit tested either.